src/main/java/org/prebid/server/bidder/impactify/ImpactifyBidder.java

```diff
@@ -0,0 +1,203 @@
+    private static final String BIDDER_CURRENCY = "USD";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+    private final CurrencyConversionService currencyConversionService;
+
+    public ImpactifyBidder(String endpointUrl, JacksonMapper mapper, CurrencyConversionService conversionService) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+        this.currencyConversionService = Objects.requireNonNull(conversionService);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<Imp> updatedImps = new ArrayList<>();
+
+        for (Imp imp : request.getImp()) {
+            BigDecimal bidFloor = imp.getBidfloor();
+            try {
+                if (shouldConvertBidFloor(bidFloor, imp.getBidfloorcur())) {
+                    bidFloor = convertBidFloorCurrency(bidFloor, request, imp.getId(), imp.getBidfloorcur());
+                }
+                updatedImps.add(updateImp(imp, bidFloor));
+            } catch (PreBidException e) {
+                return Result.withError(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        final BidRequest updatedBidRequest = request.toBuilder()
+                .imp(updatedImps)
+                .cur(List.of(BIDDER_CURRENCY))
+                .build();
+
+        return Result.withValue(HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(constructHeaders(updatedBidRequest))
+                .body(mapper.encodeToBytes(updatedBidRequest))
+                .payload(updatedBidRequest)
+                .build());
+    }
+
+    private static boolean shouldConvertBidFloor(BigDecimal bidFloor, String bidFloorCur) {
+        return bidFloor != null
+                && StringUtils.isNotEmpty(bidFloorCur)
+                && BigDecimal.ZERO.compareTo(bidFloor) <= 0
+                && !bidFloorCur.equalsIgnoreCase(BIDDER_CURRENCY);
+    }
+
+    private BigDecimal convertBidFloorCurrency(BigDecimal bidFloor, BidRequest bidRequest,
+                                               String impId, String bidFloorCur) {
+        try {
+            return currencyConversionService
+                    .convertCurrency(bidFloor, bidRequest, bidFloorCur, BIDDER_CURRENCY);
+        } catch (PreBidException e) {
+            throw new PreBidException(String.format(
+                    "Unable to convert provided bid floor currency from %s to %s for imp `%s`",
+                    bidFloorCur, BIDDER_CURRENCY, impId));
+        }
+    }
+
+    private Imp updateImp(Imp imp, BigDecimal bidFloor) {
+        return imp.toBuilder()
+                .bidfloorcur(BIDDER_CURRENCY)
+                .bidfloor(bidFloor)
+                .ext(mapper.mapper().valueToTree(ExtPrebid.of(null, parseExtImp(imp))))
```

Comment:
That's not what was done in go, check again ext processing, you can found expected result in go `sample_banner.json`